content/300-accelerate/650-troubleshoot.mdx

```diff
@@ -0,0 +1,124 @@
+
+**Suggested solution:** Consider splitting your query to fetch data in batches to prevent the 5MB limit from being reached. By splitting your query to retrieve data in batches, you ensure that each operation fetches only a portion of the total data volume, thereby not exceeding the size limit for a single fetch operation.
+
+## <inlinecode>[P6004](/orm/reference/error-reference#p6004-querytimeout)</inlinecode> (<inlinecode>QueryTimeout</inlinecode>)
+
+This error occurs when a database query fails to return a response within 10 seconds. The 10-second limit includes the duration of waiting for a connection from the pool, network latency to the database, and the execution time of the query itself. We enforce this limit to prevent unintentional long-running queries that can overload system resources.
+
+> The time for Accelerate's cross-region networking is excluded from the 10-second limit.
+
+### Possible causes for <inlinecode>[P6004](/orm/reference/error-reference#p6004-querytimeout)</inlinecode>
+
+This error could be caused by numerous reasons. Some of the prominent ones are:
+
+#### High traffic and insufficient connections
+
+If the application is receiving very high traffic and there are not a sufficient number of connections available to the database, then the queries would need to wait for a connection to become available. This situation can lead to queries waiting longer than 10 seconds for a connection, ultimately triggering a timeout error if they do not get serviced within this duration.
+
+**Suggested solution**: Review and possibly increase the `connection_limit` specified in the connection string parameter when setting up Accelerate in a platform environment ([reference](/accelerate/connection-pooling#configuring-the-connection-pool-size)).
+This limit should align with your database's maximum number of connections.
+
+By default, the connection limit is set to 10 unless a
+different `connection_limit` is specified in your database connection string.
+
+#### Long-running queries
+
+Queries may be slow to respond, hitting the 10-second timeout even when connections are available. This could happen if a very large amount of data is being fetched in a single query or if appropriate indexes are missing from the table.
+
+**Suggested solution**: Identify the slow-running queries and fetch only the required data. If you only need to use specific fields from the table, then you can use the select clause in your queries to prevent unnecessary data from being fetched. Additionally, you could consider adding appropriate indexes based on the query to ensure that data is fetched efficiently.
```

Comment:
You could also suggest to isolate long running queries into their own Environment. This way long running queries can't affect the transactional queries.